Source/AliveLibAO/Meat.cpp

```diff
@@ -346,18 +463,206 @@ void Meat::VUpdate()
+                    FP_FromDouble(0.3),
+                    36);
+                field_11C_timer = Math_NextRandom() % 16 + gnFrameCount_507670 + 60;
+            }
+            if (field_120_deadtimer < static_cast<int>(gnFrameCount_507670))
+            {
+                field_6_flags.Set(Options::eDead_Bit3);
+            }
+            break;
+
+        case 5:
+            field_B8_vely += FP_FromInteger(1);
+            field_A8_xpos += field_B4_velx;
+            field_AC_ypos += field_B8_vely;
+
+            if (!gMap_507BA8.Is_Point_In_Current_Camera_4449C0(
+                field_B2_lvl_number,
+                field_B0_path_number,
+                field_A8_xpos,
+                field_B8_vely + field_AC_ypos,
+                0))
+            {
+                field_6_flags.Set(Options::eDead_Bit3);
+            }
+            break;
+
+        default:
+            return;
+        }
+    }
 }
 
-__int16 Meat::OnCollision_438D80(BaseAliveGameObject* /*pObj*/)
+__int16 Meat::OnCollision_438D80(BaseAliveGameObject* pHit)
 {
-    NOT_IMPLEMENTED();
+    if (!pHit->field_6_flags.Get(BaseGameObject::eCanExplode_Bit7))
+    {
+        return 1;
+    }
+
+    if (pHit->field_4_typeId == Types::eMine_57 || pHit->field_4_typeId == Types::eUXB_99 || pHit->field_4_typeId == Types::eTimedMine_8)
+    {
+        return 1;
+    }
+
+    PSX_RECT bRect = {};
+    pHit->VGetBoundingRect(&bRect, 1);
+
+    if (field_114_xpos < FP_FromInteger(bRect.x) || field_114_xpos > FP_FromInteger(bRect.w))
+    {
+        field_A8_xpos -= field_B4_velx;
+        field_B4_velx = (-field_B4_velx / FP_FromInteger(2));
+    }
+    else
+    {
+        field_AC_ypos -= field_B8_vely;
+        field_B8_vely = (-field_B8_vely / FP_FromInteger(2));
+    }
+
+    pHit->VOnThrowableHit(this);
+    SFX_Play_43AE60(SoundEffect::MeatBounce_43, 0, -650, 0);
     return 0;
 }
 
 void Meat::AddToPlatform_438EA0()
 {
-    NOT_IMPLEMENTED();
+    const FP scale = field_BC_sprite_scale - FP_FromDouble(0.5);
+
+    PathLine* pLine = nullptr;
+    FP hitX = {};
+    FP hitY = {};
+    if (sCollisions_DArray_504C6C->RayCast_40C410(
+        field_A8_xpos,
+        field_AC_ypos - FP_FromInteger(20),
+        field_A8_xpos,
+        field_AC_ypos + FP_FromInteger(20),
+        &pLine,
+        &hitX,
+        &hitY,
+        scale != FP_FromInteger(0) ? 7 : 0x70))
+    {
+        if (pLine->field_8_type == 32 || pLine->field_8_type == 36)
+        {
+            for (int i = 0; i < ObjListPlatforms_50766C->Size(); i++)
+            {
+                BaseGameObject* pObjIter = ObjListPlatforms_50766C->ItemAt(i);
+                if (!pObjIter)
+                {
+                    break;
+                }
+
+                if (pObjIter->field_4_typeId == Types::eLiftPoint_51 || pObjIter->field_4_typeId == Types::eTrapDoor_98)
+                {
+                    auto pIteratedLiftPoint = static_cast<LiftPoint*>(pObjIter);
```

Comment:
should probabably cast to PlatformBase* since it can be a lift point or a trap door